src/store/ducks/opensrp/assignments/index.ts

```diff
@@ -0,0 +1,159 @@
+interface FetchAssignmentsAction extends AnyAction {
+  assignmentsByPlanId: { [key: string]: Assignment[] };
+  type: typeof ASSIGNMENTS_FETCHED;
+}
+
+/** interface for action that removes assignments from store */
+interface RemoveAssignmentsAction extends AnyAction {
+  assignmentsByPlanId: { [key: string]: Assignment[] };
+  type: typeof REMOVE_ASSIGNMENTS;
+}
+
+/** single type for all action types */
+type AssignmentActionTypes = FetchAssignmentsAction | RemoveAssignmentsAction | AnyAction;
+
+/** interface describing the assignmentsByPlanId Store */
+interface AssignmentsByPlanId {
+  [key: string]: Assignment[];
+}
+/** interface for Assignments state in store */
+interface AssignmentsStoreState {
+  assignmentsByPlanId: AssignmentsByPlanId;
+}
+
+// immutable assignments state in dux
+export type ImmutableAssignmentsStoreState = AssignmentsStoreState &
+  SeamlessImmutable.ImmutableObject<AssignmentsStoreState>;
+
+/** initial state for Assignments records in store */
+const initialOrgsStoreState: ImmutableAssignmentsStoreState = SeamlessImmutable({
+  assignmentsByPlanId: {},
+});
+
+/** the Assignment reducer function */
+export default function reducer(state = initialOrgsStoreState, action: AssignmentActionTypes) {
+  switch (action.type) {
+    case ASSIGNMENTS_FETCHED:
+      return SeamlessImmutable({
+        ...state,
+        assignmentsByPlanId: {
+          ...state.assignmentsByPlanId,
+          ...action.assignmentsByPlanId,
+        },
+      });
+    case REMOVE_ASSIGNMENTS:
+      return SeamlessImmutable({
+        ...state,
+        assignmentsByPlanId: action.assignmentsByPlanId,
+      });
+    default:
+      return state;
+  }
+}
+
+/** action to remove assignments form store */
+export const removeAssignmentsAction: RemoveAssignmentsAction = {
+  assignmentsByPlanId: {},
+  type: REMOVE_ASSIGNMENTS,
+};
+
+// action creators
+
+/** creates action to add fetched assignments to store
+ * @param {Assignment []} assignmentsList - array of assignments to be added to store
+ *
+ * @returns {FetchAssignmentsAction} - action with assignments payload that is added to store
+ */
+export const fetchAssignments = (assignmentsList: Assignment[]): FetchAssignmentsAction => {
+  const assignmentsByPlanId: AssignmentsByPlanId = {};
+
+  for (const assignment of assignmentsList) {
+    if (!assignmentsByPlanId[assignment.plan]) {
+      assignmentsByPlanId[assignment.plan] = [];
+    }
+    assignmentsByPlanId[assignment.plan].push(assignment);
+  }
+
+  return {
+    assignmentsByPlanId,
+    type: ASSIGNMENTS_FETCHED,
+  };
+};
+
+/** creates action to reset plan
+ * @param {AssignmentsByPlanId} assignmentsByPlanId object with updated assignment arrays, keyed by planId
+ * @returns {FetchAssignmentsAction}
+ */
+export const resetPlanAssignments = (
+  assignmentsByPlanId: AssignmentsByPlanId = {}
+): FetchAssignmentsAction => {
+  return {
+    assignmentsByPlanId,
+    type: ASSIGNMENTS_FETCHED,
+  };
+};
```

Comment:
New action creator to reset `assignmentsByPlanId`, this is a work around to the fact that `fetchAssignments` cannot reset anything in the store with an empty array.